pkg/backend/display/json.go

```diff
@@ -85,7 +85,7 @@ func stateForJSONOutput(s *resource.State, opts Options) *resource.State {
 	return resource.NewState(s.Type, s.URN, s.Custom, s.Delete, s.ID, inputs,
 		outputs, s.Parent, s.Protect, s.External, s.Dependencies, s.InitErrors, s.Provider,
 		s.PropertyDependencies, s.PendingReplacement, s.AdditionalSecretOutputs, s.Aliases, &s.CustomTimeouts,
-		s.ImportID, s.RetainOnDelete, s.DeletedWith)
```

Comment:
This ain't on you to fix, but I'm really hurt by the number of positional arguments here. At some point I'd have thought we'd switch over to a single struct argument.

Reply:
structs (in go) don't force you to think about new properties, most of the time these are required and it would be a bug if we forgot to set them. Having them as parameters gives us a compile error in those cases.

Reply:
I agree it's a little hairy, but the compiler did help me spot all the places I needed to consider plumbing these through 🤷

Reply:
Not the place to have a longer discussion on this, but I, too, would prefer a single struct argument here.
I agree with the general idea of wanting compiler-enforced guarantees, but I'll also admit that that's sometimes difficult to attain in Go without trading readability.
Between compiler-enforced guarantees and readability, I might pick the latter—depending on the severity of the issues that could arise.